plugins/woocommerce/src/Internal/Admin/Reviews.php

```diff
@@ -111,6 +118,233 @@ public function is_reviews_page() : bool {
+			wp_die( -1 );
+		}
+
+		$review = get_comment( $comment_id );
+
+		// Bail silently if this is not a review, or a reply to a review. That allows `wp_ajax_edit_comment()` to handle any further actions.
+		if ( ! $this->is_review_or_reply( $review ) ) {
+			return;
+		}
+
+		if ( empty( $review->comment_ID ) ) {
+			wp_die( -1 );
+		}
+
+		if ( empty( $_POST['content'] ) ) {
+			wp_die( esc_html__( 'Error: Please type your review text.', 'woocommerce' ) );
+		}
+
+		if ( isset( $_POST['status'] ) ) {
+			$_POST['comment_status'] = sanitize_text_field( wp_unslash( $_POST['status'] ) );
+		}
+
+		$updated = edit_comment();
+		if ( is_wp_error( $updated ) ) {
+			wp_die( esc_html( $updated->get_error_message() ) );
+		}
+
+		$position = isset( $_POST['position'] ) ? (int) sanitize_text_field( wp_unslash( $_POST['position'] ) ) : -1;
+		$wp_list_table = $this->make_reviews_list_table();
+
+		ob_start();
+		$wp_list_table->single_row( $review );
+		$review_list_item = ob_get_clean();
+
+		$x = new WP_Ajax_Response();
+
+		$x->add(
+			array(
+				'what'     => 'edit_comment',
+				'id'       => $review->comment_ID,
+				'data'     => $review_list_item,
+				'position' => $position,
+			)
+		);
+
+		$x->send();
+	}
+
+	/**
+	 * Ajax callback for replying to a review inline. This is based on {@see wp_ajax_replyto_comment()}, and is designed
+	 * to run before that callback so we can override the rendering for a review reply.
+	 *
+	 * @return void
+	 */
+	public function handle_reply_to_review(): void {
+		check_ajax_referer( 'replyto-comment', '_ajax_nonce-replyto-comment' );
+
+		$reply_post_id = isset( $_POST['comment_post_ID'] ) ? (int) sanitize_text_field( wp_unslash( $_POST['comment_post_ID'] ) ) : 0;
+		$post          = get_post( $reply_post_id );
+
+		if ( ! $post ) {
+			wp_die( -1 );
+		}
+
+		// Inline Review replies will use the `detail` mode. If that's not what we have, then let WordPress core take over.
+		if ( isset( $_REQUEST['mode'] ) && 'dashboard' === $_REQUEST['mode'] ) {
+			return;
+		}
+
+		// If this is not a a reply to a review, bail silently to let WordPress core take over.
+		if ( 'product' !== get_post_type( $post ) ) {
+			return;
+		}
+
+		if ( ! current_user_can( 'edit_post', $reply_post_id ) ) {
+			wp_die( -1 );
+		}
+
+		if ( empty( $post->post_status ) ) {
+			wp_die( 1 );
+		} elseif ( in_array( $post->post_status, array( 'draft', 'pending', 'trash' ), true ) ) {
+			wp_die( esc_html__( 'Error: You can\'t reply to a review on a draft product.', 'woocommerce' ) );
+		}
+
+		$user = wp_get_current_user();
+
+		if ( $user->exists() ) {
+			$user_ID              = $user->ID;
+			$comment_author       = wp_slash( $user->display_name );
+			$comment_author_email = wp_slash( $user->user_email );
+			$comment_author_url   = wp_slash( $user->user_url );
+			$comment_content      = isset( $_POST['content'] ) ? wp_unslash( $_POST['content'] ) : ''; // phpcs:ignore WordPress.Security.ValidatedSanitizedInput.InputNotSanitized
```

Comment:
WordPress core doesn't sanitize this in `wp_ajax_replyto_comment()`. I think it's sanitized somewhere in the pre-insert filters.

Reply:
I'd suggest adding an inline comment to justify `phpcs:ignore WordPress.Security.ValidatedSanitizedInput.InputNotSanitized` considering that this code will be reviewed by the WC team in the future. That would avoid some back and forth to explain why we're suppressing this check.